### ISSUE TITLE
Windows version of ykcs11 segfaults

### ISSUE THREAD

**Issue by denisenkom** [Yubico/yubico-piv-tool#59]
this command fails:

pkcs11-tool --module ykcs11.dll -L

immediately after C_GetFunctionList call

I suspect that Windows build does not set packing correctly for pkcs11.h, it should set it to 1 byte alignment.  From pkcs11.h:
```
 * In addition to defining these 6 macros, the packing convention
 * for PKCS #11 structures should be set.  The PKCS #11
 * convention on packing is that structures should be 1-byte
 * aligned.
 *
 * In a Win32 environment, this might be done by using the
 * following preprocessor directive before including pkcs11.h
 * or pkcs11t.h:
 *
 * #pragma pack(push, cryptoki, 1)
 *
 * and using the following preprocessor directive after including
 * pkcs11.h or pkcs11t.h:
 *
 * #pragma pack(pop, cryptoki)
```

**Comment by a-dma** [comment]
I've spent some time on this.
The fix seems pretty straightforward, however it does something weird on a 64bit toolchain (pkcs11-tool + ykcs11).
The call to SCardListReaders in ykpiv.c:233 corrupts the first 4 bytes of the buffer. I believe that the unwanted data are the four MSB of the little-endian representation of the pointer, but I might be wrong. Anyway I think this has something to do with packaging and pointer sizes in 32 vs 64 bits but at this stage I'm unsure whether the  issue is in ykpiv, ykcs11 or pkcs11-tool.

The 32-bit version should however work. There is no release of this yet, so building from sources is required (I can provide a .zip with the built version if needed).

**Comment by denisenkom** [comment]
I am sending pull request: https://github.com/Yubico/yubico-piv-tool/pull/62.  I tested it on Win64.  I also fixes couple additional problems in Windows.

**Comment by a-dma** [comment]
Closed via #62.
